### ISSUE TITLE
Add error in return value of constructors for RepositoryStorage backends

### ISSUE THREAD

**Issue by nkashy1** [doc-ai/tensorio-models#19]
Constructors for `RepositoryStorage` objects should returns errors as well as `RepositoryStorage` structs -- this gives us ability to do more intelligent healthz checks

This will be important when we implement a `RepositoryStorage` backend that talks to a database, but it would be a good idea to do this also with the memory backend.

**Comment by fkautz** [comment]
I don't think this is appropriate, we cannot predict ahead of time what parameters a RepositoryStorage backend will take. 

e.g.
NewMemory() RepositoryStorage
NewGCS(connection string) RepositoryStorage, error
NewFilesystem(dir Dir) RepositoryStorage, error
NewDB(ip IP, port int, user, password string) RepositoryStorage, error
NewWebKV(addr, authToken string) RepositoryStorage, error
